String.cxx

```diff
@@ -119,9 +121,9 @@ class String {
 
     inline void setCharAt(unsigned index, const char* c) { m_u[index] = c; }
 
-    inline String substring(unsigned from) { return m_u.substr(from); }
+    [[nodiscard]] inline String substring(unsigned from) const noexcept { return m_u.substr(from); }
 
-    inline String substring(unsigned from, unsigned to){ return m_u.substr(from, to - from); }
+    [[nodiscard]] inline String substring(unsigned from, unsigned to) const noexcept { return m_u.substr(from, to - from); }
```

Comment:
Not `noexcept`; could throw `std::bad_alloc`

Reply:
Did not consider that, thanks